src/components/collection/Picker.tsx

```diff
@@ -1,85 +1,189 @@
     const collectionsLabel = useConstant(() =>
         intl.formatMessage({
             id: 'entityCreate.bindingsConfig.collectionsLabel',
         })
     );
+
+    const [collectionData, setCollectionData] = useState<CollectionData[]>([]);
     const [missingInput, setMissingInput] = useState(false);
 
-    const { liveSpecs: collectionData, error } = useLiveSpecs('collection');
+    const {
+        liveSpecs,
+        error: liveSpecsError,
+        isValidating: isValidatingLiveSpecs,
+    } = useLiveSpecs('collection');
+
+    // Draft Editor Store
+    const persistedDraftId = useEditorStore_persistedDraftId();
 
     // Form State Store
-    const messagePrefix = useFormStateStore_messagePrefix();
+    const formActive = useFormStateStore_isActive();
 
     // Resource Config Store
     const collections = useResourceConfig_collections();
+    const discoveredCollections = useResourceConfig_discoveredCollections();
+    const setRestrictedDiscoveredCollections =
+        useResourceConfig_setRestrictedDiscoveredCollections();
 
     const setResourceConfig = useResourceConfig_setResourceConfig();
 
+    const {
+        draftSpecs,
+        error: draftSpecsError,
+        isValidating: isValidatingDraftSpecs,
+    } = useDraftSpecs(persistedDraftId);
+
+    useEffect(() => {
+        const populateCollectionData =
+            entityType === ENTITY.MATERIALIZATION
+                ? liveSpecs.length > 0
+                : !isValidatingLiveSpecs && !isValidatingDraftSpecs;
+
+        if (populateCollectionData) {
+            let collectionsOnServer: CollectionData[] = liveSpecs.map(
+                ({ catalog_name }) => ({
+                    name: catalog_name,
+                    classification: 'Existing Collections',
+                })
+            );
+
+            if (entityType === ENTITY.CAPTURE) {
+                collectionsOnServer = [
+                    ...draftSpecs
+                        .filter(
+                            ({ spec_type }) => spec_type === ENTITY.COLLECTION
+                        )
+                        .map(({ catalog_name }) => ({
+                            name: catalog_name,
+                            classification: 'Discovered Collections',
+                        })),
+                    ...collectionsOnServer,
+                ];
+            }
+
+            setCollectionData(collectionsOnServer);
+        }
+    }, [
+        setCollectionData,
+        collections,
+        draftSpecs,
+        entityType,
+        isValidatingDraftSpecs,
+        isValidatingLiveSpecs,
+        liveSpecs,
+        workflow,
+    ]);
+
     const handlers = {
-        updateCollections: (event: React.SyntheticEvent, value: any) => {
-            setResourceConfig(value);
+        updateCollections: (
+            event: React.SyntheticEvent,
+            value: CollectionData[],
+            reason: AutocompleteChangeReason
+        ) => {
+            const removeOptionWithBackspace =
```

Comment:
This boolean detects whether the user is attempting to remove one of the selected autocomplete options via the backspace key. While this feels a little hacky, it is MUI's recommended solution to this problem.

The same can be said of the logic used to prevent tags from rendering in an autocomplete component with the `multiple` prop set. MUI advises one to supply a function that returns null to the `renderTags` prop to achieve this.